ui/server.py

```diff
@@ -417,6 +424,63 @@ def update_discovery_group():
         return 'OK', 200
 
 
+def assign_categories(discoveries):
```

Comment:
please turn this method to "private", i.e., call it `_assign_categories`. In python there is no visibility like public/protected/private in Java and other programming languages, but a `_` before the name of the method is used by developer to indicate it's "private"